core/chains/evm/client/node.go

```diff
@@ -343,17 +343,19 @@ func (n *node) verify(callerCtx context.Context) (err error) {
 
 func (n *node) Close() error {
 	return n.StopOnce(n.name, func() error {
-		defer n.wg.Wait()
+		defer func() {
+			n.wg.Wait()
+			if n.ws.rpc != nil {
+				n.ws.rpc.Close()
+			}
+		}()
 
 		n.stateMu.Lock()
 		defer n.stateMu.Unlock()
 
 		n.cancelNodeCtx()
 		n.cancelInflightRequests()
 		n.state = NodeStateClosed
-		if n.ws.rpc != nil {
```

Comment:
Reading at this point races with writes from the `dial()` method, which we can't be sure is done running until after `n.wg.Wait()` returns.